ui/imports/utils/Global.qml

```diff
@@ -52,70 +49,15 @@ Item {
     signal openEditDisplayNamePopup()
     signal openActivityCenterPopupRequested
 
-    function openContactRequestPopup(publicKey) {
-        const contactDetails = Utils.getContactDetailsAsJson(publicKey);
-        return openPopup(sendContactRequestPopupComponent, {
-            userPublicKey: publicKey,
-            userDisplayName: contactDetails.displayName,
-            userIcon: contactDetails.largeImage,
-            userIsEnsVerified: contactDetails.ensVerified
-        })
-    }
+    signal openContactRequestPopup(string publicKey)
```

Comment:
Thank you @igor-sirotin for the review and vigilance! I totally missed the fact that we have returns here :man_facepalming: So it's necessary to totally rethink the idea. What comes to my mind is to pass callback as an additional argument. It would be
```
signal openContactRequestPopup(string publicKey, var cb)
```
in this case.

On the caller side it would be e.g.:
```
                Global.openContactRequestPopup(root.publicKey, popup => {
                    popup.accepted.connect(d.reload)
                })
```
The drawback is a slightly more complicated call. But there is an advantage too: we can easily refactor implementation to create popups asynchronously and therefore trim startup time. It's not straightforward in current approach where popup is returned immediately. @igor-sirotin @caybro @osmaczko what do you think?

Reply:
I doubt `Promise` is supported by QML JS engine but worth exploring

Reply:
yeah, the convenience is a drawback here, I know... But possibility of making it async is a quite good reason to use it regardless. I would skip handling errors on caller side. Caller shouldn't bother about potential problems with loading and creating components and just assume that popup is created and provided via the callback. When it comes to `Promise` - it's not supported out of the box, there are some libs for that in qml, but it's not worth to use them for this case imo.

Reply:
Ahh! Sorry! Promises are supported. I've been using standard js promises in qml. So we could use them as well. I will do further research about that and propose sth.

Reply:
But wait. Promise would have to be returned in normal way, but it's not possible for signals :facepalm:

Reply:
> The drawback is a slightly more complicated call. But there is an advantage too: we can easily refactor implementation to create popups asynchronously and therefore trim startup time. It's not straightforward in current approach where popup is returned immediately. @igor-sirotin @caybro @osmaczko what do you think?

Callback solutions sound ok at the first sight. I don't have any better idea in mind.